e2e/_suites/metricbeat/metricbeat_test.go

```diff
@@ -378,14 +378,14 @@ func (mts *MetricbeatTestSuite) runMetricbeatService() error {
 			return err
 		}
 
-		err = docker.LoadImage(imagePath)
+		err = deploy.LoadImage(imagePath)
 		if err != nil {
 			return err
 		}
 
 		mts.Version = mts.Version + "-amd64"
 
-		err = docker.TagImage(
+		err = deploy.TagImage(
```

Comment:
Same here, abstract out tagging an image to support both docker/k8s and keep docker's version of TagImage in its own namespace

Reply:
Yep, I totally agree with you about the docker namespace would be the right thing to do. But because of the ServiceRequest struct, which resides in the deploy namespace, and how `deploy` package calls the `docker` one, it was not possible to define the ServiceRequest struct inside the `deploy` package without a cyclic reference.

Reply:
I did that as a compromise to add progress. Maybe the Provider.Client struct would solve this issue, which we could discuss/face it in follow up PRs

Reply:
Yep makes sense now!